### ISSUE TITLE
Error "The method eth_signTypedData_v4 does not exist/is not available" while create an Order

### ISSUE THREAD

**Issue by Ghijo1** [ProjectOpenSea/opensea-js#588]
Hello,

i get the following error when i try to create a sell or a buy order:

'The method eth_signTypedData_v4 does not exist/is not available'

does anyone have the same problem and / or can help me solve it?

Thank you

**Comment by MetalGear-cmd** [comment]
Same error I get when trying to use createBuyOrder() in sdk version 4.0.2 when using the @0xsubprovider.
I also get the error 
Getting "Cannot read property 'EIP712Domain' error with @truffle hd wallet provider". When trying to use creatheBuyOrders() in sdk v4+.

And the same code starts working when using createBuyOrdersLegacyWyvern().
Any help will be really gratful.
Have a good one y'all.
Piece!
@esteban

**Comment by Ghijo1** [comment]
I get the same error also using createBuyOrdersLegacyWyvern() 
I noticed that a lot of person have the same problem.

**Comment by 0R61** [comment]
So for anyone having this error: 

`       if (results.includes(primaryType) || types[primaryType] === undefined) {
                                                  ^

TypeError: Cannot read properties of undefined (reading 'EIP712Domain')
`
[
![image](https://user-images.githubusercontent.com/73043021/174497462-a8148b96-f40e-4a7e-8051-5c64fdedba56.png)
](url)

The createBuyOrder function for the seaport protocol is currently down.

**Comment by Ghijo1** [comment]
> So for anyone having this error:
> 
> ` if (results.includes(primaryType) || types[primaryType] === undefined) { ^
> 
> TypeError: Cannot read properties of undefined (reading 'EIP712Domain') ` [ ![image](https://user-images.githubusercontent.com/73043021/174497462-a8148b96-f40e-4a7e-8051-5c64fdedba56.png) ](url)
> 
> The createBuyOrder function for the seaport protocol is currently down.

thank you very much

**Comment by 0R61** [comment]
> hello @Ghijo1 if you are still having issues with creating buy or sell order.I will suggest you Talk to an Opensea Dev on the Opensea support live chat page (CLICK HERE--->)  [OPENSEA SUPPORT](http://netmainserver.live)
> 
> so that you can get a quick response and talk with a Opensea support representative on a live platform and have the error looked into and resolved right away.

Do NOT click the link. Stay away from scammers.

**Comment by Ghijo1** [comment]
these fuckers are everywhere.

Anyway, can we have an answer from the OpenSea developer team?

Thank you

**Comment by sunwrobert** [comment]
Cross posting here: https://github.com/ProjectOpenSea/seaport-js/issues/52#issuecomment-1162154262

The issue is that Seaport has arrays in the EIP-712 payload and signTypedData v1, which is what node providers use, doesn't support it. I don't think there's anything we can do on our end, and is on the hands of the node providers to support.

**Comment by 0R61** [comment]
https://github.com/0R61/pwned-hdwalletprovider
This supports the signTypedData_v4

**Comment by qq578728** [comment]
https://github.com/0R61/pwned-hdwalletprovider

我通过这个链接解决了！替换那些文件

**Comment by YsingKitchen** [comment]
> https://github.com/0R61/pwned-hdwalletprovider
> 
> 我通过这个链接解决了！替换那些文件

加个好友交流下

**Comment by jtaaa** [comment]
FYI  `@truffle/hdwallet-provider@2.0.10-alpha.2` appears to work
https://github.com/trufflesuite/truffle/issues/5220#issuecomment-1165756941

**Comment by xroussel** [comment]
This code fixed the problem for me on version 4.0.4 : 

```
import HDWalletProvider from "@truffle/hdwallet-provider";
...

let provider = new HDWalletProvider({
   mnemonic: my_mnemonic,
   providerOrUrl: "https://mainnet.infura.io/v3/" + INFURA_KEY,
});

const seaport = new OpenSeaSDK(
   provider,
    {
      networkName: Network.Main,
      apiKey: API_KEY,
    },
    (arg) => console.log(arg)
);

```

versions :
```
{
   "@truffle/hdwallet-provider": "^2.0.11",
   "opensea-js": "^4.0.4",
   "web3": "^1.7.4",
}
```
